### PR TITLE
#2820 Grant private access

### PR DESCRIPTION
**Pull request by x1c0**

### CODE REVIEW

File: apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts [hunk 1]
```diff
@@ -26,13 +27,15 @@ export class CreateOrUpdateAccessDialog implements OnDestroy {
   public constructor(
     @Inject(MAT_DIALOG_DATA) public data: CreateOrUpdateAccessDialogParams,
     public dialogRef: MatDialogRef<CreateOrUpdateAccessDialog>,
-    private formBuilder: FormBuilder
+    private formBuilder: FormBuilder,
+    private dataService: DataService
```
Comment (dtslvr) [inline apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts:31]: Can you move to the second position to keep the alphabetical order?

File: apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts [hunk 2]
```diff
@@ -43,10 +46,20 @@ export class CreateOrUpdateAccessDialog implements OnDestroy {
   public onSubmit() {
     const access: CreateAccessDto = {
       alias: this.accessForm.controls['alias'].value,
-      type: this.accessForm.controls['type'].value
+      type: this.accessForm.controls['type'].value,
+      granteeUserId: this.accessForm.controls['userId'].value
     };
 
-    this.dialogRef.close({ access });
+    this.dataService.postAccess(access).subscribe({
```
Comment (dtslvr) [inline apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts:53]:
```
this.dataService
.postAccess(access)
.pipe(
  catchError((error) => {
    if (error.status === StatusCodes.BAD_REQUEST) {
      alert($localize`Oops! Could not grant access.`);
    }
    
    return EMPTY;
  }),
  takeUntil(this.unsubscribeSubject))
.subscribe(() => {
  this.dialogRef.close();
});
```

File: apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts [hunk 3]
```diff
@@ -26,13 +27,15 @@ export class CreateOrUpdateAccessDialog implements OnDestroy {
...
+    private formBuilder: FormBuilder,
+    private dataService: DataService
   ) {}
 
   ngOnInit() {
     this.accessForm = this.formBuilder.group({
       alias: [this.data.access.alias],
-      type: [this.data.access.type, Validators.required]
+      type: [this.data.access.type, Validators.required],
+      userId: [this.data.access.grantee, Validators.required]
```
Comment (dtslvr) [inline apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts:38]: Is this working with the `*ngIf`? (required even if it's not there)

File: apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.html [hunk 4]
```diff
@@ -22,9 +22,21 @@ <h1 i18n mat-dialog-title>Grant access</h1>
         <mat-label i18n>Type</mat-label>
         <mat-select formControlName="type">
           <mat-option i18n value="PUBLIC">Public</mat-option>
+          <mat-option i18n value="PRIVATE">Private</mat-option>
         </mat-select>
       </mat-form-field>
     </div>
+    <div *ngIf="accessForm.controls['type'].value === 'PRIVATE' ">
```
Comment (dtslvr) [inline apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.html:29]: You can use the [new control flow](https://angular.dev/guide/templates/control-flow#if-block-conditionals) of Angular:
```
@if (accessForm.controls['type'].value === 'PRIVATE') {
  <div>
    ...
  </div>
}
```

File: apps/client/src/app/components/user-account-access/user-account-access.component.ts [hunk 5]
```diff
@@ -112,25 +112,9 @@ export class UserAccountAccessComponent implements OnDestroy, OnInit {
...
-      .subscribe((data: any) => {
-        const access: CreateAccessDto = data?.access;
-
-        if (access) {
-          this.dataService
-            .postAccess({ alias: access.alias })
-            .pipe(takeUntil(this.unsubscribeSubject))
-            .subscribe({
-              next: () => {
-                this.update();
```
Comment (dtslvr) [inline apps/client/src/app/components/user-account-access/user-account-access.component.ts:127]: Is the table still updated after granting access?

File: libs/common/src/lib/interfaces/access.interface.ts [hunk 6]
```diff
@@ -1,6 +1,6 @@
 export interface Access {
   alias?: string;
-  grantee: string;
+  grantee?: string;
   id: string;
-  type: 'PUBLIC' | 'RESTRICTED_VIEW';
+  type: 'PUBLIC' | 'PRIVATE' | 'RESTRICTED_VIEW';
```
Comment (dtslvr) [inline libs/common/src/lib/interfaces/access.interface.ts:5]: Please move `PRIVATE` to the first position

File: CHANGELOG.md [hunk 7]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 - Improved the user interface of the access table to share the portfolio
+- Grant private access
```
Comment (dtslvr) [inline CHANGELOG.md:13]: `Added support to grant private access` 😉

**Review by dtslvr — CHANGES_REQUESTED** [review]
Very quick implementation of this nice feature 😎
I have added a few comments.

File: apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts (hunk 3 above)
Comment (x1c0) [reply apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts:38]: ah is not :) ill fix it

File: apps/client/src/app/components/user-account-access/user-account-access.component.ts (hunk 5 above)
Comment (x1c0) [reply apps/client/src/app/components/user-account-access/user-account-access.component.ts:127]: no is not.. only after refresh, ill fix it

**Review by dtslvr — APPROVED** [review]
Great addition, @x1c0! 😎
Ready to merge now.

**Comment by dtslvr** [comment]
Follow-up ticket: https://github.com/ghostfolio/ghostfolio/issues/2830

**Pull request merged**